pyomo/contrib/community_detection/detection.py

```diff
@@ -0,0 +1,156 @@
+        created from the Pyomo model
+        file_destination: an optional argument that takes in a path if the user wants to save an edge and adjacency
+        list based on the model
+        log_level: determines the minimum severity of an event for it to be included in the event logger file; can be
+        specified as any of the following (in order of increasing severity): logging.DEBUG, logging.INFO,
+        logging.WARNING, logging.ERROR, logging.CRITICAL; These levels correspond to integer values of 10, 20, 30, 40,
+        and 50 (respectively). Thus, log_level can also be specified as an integer.
+        random_seed : takes in an integer to use as the seed number for the heuristic Louvain community detection
+
+    Returns:
+        community_map: a Python dictionary whose keys are integers from zero to the number of communities minus one
+        with values that are sorted lists of the nodes in the given community
+    """
+
+    # Use this function as a check to make sure all of the arguments are of the correct type, else return None
+    if check_for_correct_arguments(model, node_type, with_objective, weighted_graph, file_destination, log_level,
+                                   random_seed) is False:
+        return None
+
+    # Generate the model_graph (a networkX graph) based on the given Pyomo optimization model
+    model_graph = community_graph._generate_model_graph(model, node_type=node_type, with_objective=with_objective,
+                                                        weighted_graph=weighted_graph,
+                                                        file_destination=file_destination)
+
+    # Use Louvain community detection to determine which community each node belongs to
+    partition_of_graph = community.best_partition(model_graph, random_state=random_seed)
+
+    # Use partition_of_graph to create a dictionary that maps communities to nodes (because Louvain community detection
+    # returns a dictionary that maps individual nodes to their communities)
+    number_of_communities = len(set(partition_of_graph.values()))
+    community_map = {nth_community: [] for nth_community in range(number_of_communities)}
+    for node in partition_of_graph:
+        nth_community = partition_of_graph[node]
+        community_map[nth_community].append(node)
+
+    # Log information about the number of communities found from the model
+    logging.info("%s communities were found in the model" % number_of_communities)
+    if number_of_communities == 0:
+        logging.error("in detect_communities: Empty community map was returned")
+    if number_of_communities == 1:
+        logging.warning("Community detection found that with the given parameters, the model could not be decomposed - "
+                        "only one community was found")
+
+    return community_map
+
+
+def check_for_correct_arguments(model, node_type, with_objective, weighted_graph, file_destination, log_level,
```

Comment:
Since this is some sort of internal function with a generic name I would suggest renaming it as `_check_for_correct_arguments`